### ISSUE TITLE
False positive in javascript.lang.best-practice.assigned-undefined.assigned-undefined

### ISSUE THREAD

**Issue by sk-** [returntocorp/sgrep-rules#1280]
**Describe the bug**
assigning `undefined` to any variable or constant will trigger the rule in some contexts, even when it's safe to use undefined. What the rule is trying to prevent is to overwrite the actual value of `undefined`. 

**To Reproduce**
```js
function foo() {
  const a = undefined;
  let b = undefined;
  b = undefined;
  var c = undefined;
}

const a = undefined;
let b = undefined; // No error
b = undefined; // No error
var c = undefined; // No error
```

**Expected behavior**
None of the lines above should be trigger an error of the rule.

**Priority**
How important is this to you?
- [ ] P0: blocking me from making progress
- [X] P1: this will block me in the near future
- [ ] P2: annoying but not blocking me

**Additional Context**
I think what's going on is that in some contexts the fist rule (`undefined = $X;`) is being interpreted as symmetric, hence matching in the opposite direction.

See the output below:

```
semgrep -l js -e 'undefined = $X;' test.js
test.js
2:  const a = undefined;
--------------------------------------------------------------------------------
3:  let b = undefined;
--------------------------------------------------------------------------------
4:  b = undefined;
--------------------------------------------------------------------------------
5:  var c = undefined;
--------------------------------------------------------------------------------
8:const a = undefined;
ran 1 rules on 1 files: 5 findings
```

**Comment by inkz** [comment]
:pray: looks like this issue is solved in current version of semgrep https://semgrep.dev/s/e4vl/
